grblas/ops.py

```diff
@@ -590,39 +642,31 @@ def register_new(cls, name, monoid, binaryop):
         setattr(module, funcname, semiring)
 
 
-def find_opclass(gb_op):
-    if isinstance(gb_op, (OpBase, ParameterizedUdf)):
-        opclass = gb_op.__class__.__name__
+def get_typed_op(op, dtype, dtype2=None):
+    if isinstance(op, OpBase):
+        if dtype2 is not None:
+            dtype = dtypes.unify(dtype, dtype2)
+        return op[dtype]
+    elif isinstance(op, ParameterizedUdf):
+        op = op()  # Use default parameters of parameterized UDFs
+        return get_typed_op(op, dtype, dtype2)
+    elif isinstance(op, TypedOpBase):
+        return op
     else:
-        for opclass in (UnaryOp, BinaryOp, Monoid, Semiring):
-            if gb_op in opclass.all_known_instances:
-                opclass = opclass.__name__
-                break
-        else:
-            opclass = UNKNOWN_OPCLASS
-    if opclass.startswith('Parameterized'):
-        gb_op = gb_op()  # Use default parameters of parameterized UDFs
-        return find_opclass(gb_op)
-    return gb_op, opclass
-
-
-def reify_op(gb_op, dtype, dtype2=None):
-    if dtype2 is not None:
-        dtype = dtypes.unify(dtype, dtype2)
-    if isinstance(gb_op, OpBase):
-        gb_op = gb_op[dtype]
-    return gb_op
-
-
-_return_type = {}
+        raise TypeError(f'Unable to get typed operator from object with type {type(op)}')
 
 
-def find_return_type(gb_op):
+def find_opclass(gb_op):
     if isinstance(gb_op, OpBase):
-        raise ValueError('Requires concrete operator. Call `reify_op` first.')
-    if gb_op not in _return_type:
-        raise KeyError('Unknown operator. You must register function prior to use.')
-    return _return_type[gb_op]
+        opclass = gb_op.__class__.__name__
+    elif isinstance(gb_op, TypedOpBase):
+        opclass = gb_op.opclass
+    elif isinstance(gb_op, ParameterizedUdf):
+        gb_op = gb_op()  # Use default parameters of parameterized UDFs
+        gb_op, opclass = find_opclass(gb_op)
+    else:
+        opclass = UNKNOWN_OPCLASS
+    return gb_op, opclass
```

Comment:
Is this function used anywhere other than in the tests? If not, we should remove it and refactor the tests to use `get_typed_op` instead.

Reply:
It's used one place: `grblas.base:GbContainer.__call__` to determine whether an argument in `*optional_mask_and_accum` is a binary op for accum.  I was hoping to remove `find_opclass` as well.